core/src/test/java/kafka/security/JaasTestUtils.java

```diff
@@ -0,0 +1,496 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package kafka.security;
+
+import kafka.utils.TestUtils;
+
+import org.apache.kafka.clients.admin.ScramMechanism;
+import org.apache.kafka.common.config.SaslConfigs;
+import org.apache.kafka.common.utils.Java;
+
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Properties;
+
+import static org.apache.kafka.common.config.SaslConfigs.GSSAPI_MECHANISM;
+import static org.apache.kafka.common.security.oauthbearer.OAuthBearerLoginModule.OAUTHBEARER_MECHANISM;
+import static org.apache.kafka.common.security.plain.internals.PlainSaslServer.PLAIN_MECHANISM;
+
+public class JaasTestUtils {
+
+    public static class Krb5LoginModule implements JaasModule {
```

Comment:
Could you please move those impls of `JaasModule` to separate class? for example:
```java
public class JaasModule {
    public static JaasModule zkDigestModule(boolean debug, Map<String, String> entries) {
        String name = "org.apache.zookeeper.server.auth.DigestLoginModule";
        return new JaasModule(
            name,
            debug,
            entries,
            String.format("%s required\n  debug=%b\n  %s;\n", name, debug, entries.entrySet().stream()
                .map(e -> e.getKey() + "=\"" + e.getValue() + "\"")
                .reduce((e1, e2) -> e1 + "\n  " + e2)
                .orElse(""))
        );
    }

    private final String name;

    private final boolean debug;

    private final Map<String, String> entries;
    
    private final String toString;

    private JaasModule(String name, boolean debug, Map<String, String> entries, String toString) {
        this.name = name;
        this.debug = debug;
        this.entries = entries;
        this.toString = toString;
    }

    String name() {
        return name;
    }

    boolean debug() {
        return debug;
    }

    Map<String, String> entries() {
        return entries;
    }
    
    @Override
    public String toString() {
        return toString;
    }
}
```
It seems to me that is more readable. Also, we don't need to have many class declaration.